pystac_client/item_search.py

```diff
@@ -54,8 +55,8 @@
 Sortby = List[Dict[str, str]]
 SortbyLike = Union[Sortby, str, List[str]]
 
-Fields = List[str]
-FieldsLike = Union[Fields, str]
+Fields = Dict[str, List[str]]
```

Comment:
There's enough `staticmethod` logic around `Fields` that you _could_ make the case for `Fields` to be its own class, with a `parse` classmethod and a `to_string` method. I don't think it's required, just a thought.

Reply:
I agree -- I think there's a good deal of refactoring that could be done here with validation. For now, I just wanted to get the existing patterns working correctly